horovod/run/run_func.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2018 Uber Technologies, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import print_function
+import cloudpickle
+import tempfile
+import os
+import sys
+import six
+import subprocess
+import collections
+import textwrap
+import time
+
+from horovod.run.util import network
+from horovod.run.common.util import safe_shell_exec
+
+_TAIL_LINES_TO_KEEP = 100
+
+_LOCAL_PICKLED_RESULT_FILENAME = "local_result.pkl"
+_PICKLED_RESULT_FILENAME = "result.pkl"
+_PICKLED_PROC_FN_FILENAME = "proc_fn.pkl"
+_PROC_LAUNCHER_FILENAME = "launch.sh"
+
+
+def run_func(
+        fn,
+        args=None,
+        kwargs=None,
+        num_proc=None,
+        host=None,
```

Comment:
`host` or `hosts`?

Reply:
keep the same with `horovodrun` arg `host`